lib/pbench/server/api/resources/query_apis/index_mappings.py

```diff
@@ -0,0 +1,81 @@
+from http import HTTPStatus
+from flask import jsonify
+from flask_restful import abort, Resource
+from logging import Logger
+from typing import Any, AnyStr, Dict
+
+from pbench.server import PbenchServerConfig
+from pbench.server.database.models.template import Template
+
+
+class IndexMappings(Resource):
+    """
+    Get properties of the run-data document type on which the dashboard can search.
+    This API currently only returns mapping properties of run-data documents.
+
+    TODO: In future we can tweak this API to take the document type name from the user
+    and query that type name against the template table to return the appropriate mappings
+
+    :return:
+    {
+        "@metadata":
+            [
+                "controller_dir",
+                "file-date",
+                "file-name",
+                "file-size",
+                "md5",
+                ...
+            ],
+       "host_tools_info":
+            [
+                "hostname",
+                "hostname-f",
+                ...
+            ],
+        "run":
+            [
+                "id",
+                "controller",
+                "user",
+                "name",
+                ...
+            ],
+        "sosreports":
+            [
+                "hostname-f",
+                "sosreport-error",
+                "hostname-s",
+                ...
+            ]
+    }
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.config = config
+        self.logger = logger
+
+    def get(self) -> Dict[AnyStr, Any]:
+        """
+        Return mapping properties of the run-data document by querying the template database
+        """
+        try:
+            template = Template.find("run")
+            mappings = template.mappings
+
+            result = {}
+            for property in mappings["properties"]:
+                if "properties" in mappings["properties"][property]:
+                    result[property] = list(
+                        mappings["properties"][property]["properties"].keys()
+                    )
+
+            result.pop("authorization", None)
```

Comment:
I'm still uncomfortable with "lying" like this. The more I think about it, the more I think we ought to return the complete "schema" and let the dashboard effect strip it out to avoid putting it on the field selector panel.

Reply:
@dbutenhof, would it change your mind if the client were something other than the Dashboard?  That is, does this field need to be hidden from anyone other than a Dashboard User?

Reply:
Yes, @webbnh, that's my point; it's definitely not needed for the dashboard, but I don't feel entirely comfortable about hiding it from someone else.

For example, I've had thoughts about adding an `ELASTICSEARCH` *role* that allows privileged clients to continue using the direct `/api/v1/elasticsearch` query for more advanced (and implicitly `ADMIN` class) queries; if they want to see the document templates, we probably don't want to hide this.

But I'm not entirely sure what the use case it for allowing queries entirely outside out authorization framework (even with the special role, it sounds risky), and outside of that I'm not really sure how it would benefit anyone.

Reply:
This feeds into Pete's issue about trying to generalize a "schema" API that would include our ES templates. I like that idea, but it doesn't necessarily have to invalidate this first attempt. I generally appreciated being "agile" and solving the problem in front of us and being prepared to refactor and rebuild later as we learn more. With that in mind, I think I'm going to resolve this conversation for now.